main.go

```diff
@@ -100,17 +118,16 @@ func main() {
 	log.Infof("Eject project")
 	{
 		if err := e.eject(); err != nil {
-			failf("Failed to eject project: %s", err)
+			failfDefered(func() { logout(e) }, "Failed to eject project: %s", err)
```

Comment:
I would recommend creating a "main2" function which returns errors and it is called from main. This would prevent the need of isLoggedIn and failedDefered as the "defer logout(e)" would called when the function returns.

Reply:
That is a great idea 👍